OloEditor/src/EditorLayer.cpp

```
@@ -1206,6 +1247,109 @@ namespace OloEngine
+            break;
+        }
+        if (!terrainEntity || !terrainEntity.GetComponent<TerrainComponent>().m_TerrainData)
+        {
+            return false;
+        }
+
+        auto const& tc = terrainEntity.GetComponent<TerrainComponent>();
+        auto const& transform = terrainEntity.GetComponent<TransformComponent>();
+
+        // Convert mouse position to NDC [-1, 1]
+        float ndcX = (mousePos.x / viewportSize.x) * 2.0f - 1.0f;
```

Clang (14.0.6)

[warning] variable 'ndcX' is not initialized

(cppcoreguidelines-init-variables)